safenode/src/protocol/types/wallet/mod.rs

```diff
@@ -0,0 +1,109 @@
+// Copyright 2023 MaidSafe.net limited.
+//
+// This SAFE Network Software is licensed to you under The General Public License (GPL), version 3.
+// Unless required by applicable law or agreed to in writing, the SAFE Network Software distributed
+// under the GPL Licence is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. Please review the Licences for the specific language governing
+// permissions and limitations relating to use of the SAFE Network Software.
+
+//! An implementation of a local Wallet used by clients and nodes (the latter use them for their rewards).
+//! There is one which is deposit only, and one which can also send tokens.
+//!
+//! Later, a network Wallet store can be implemented thusly:
+//! 1. Chunk each Dbc, both spent and available.
+//! 2. For a semi-public Wallet:
+//!     a. Store a register with address of your `PublicAddress`.
+//!    Then push these ops:
+//!     b. self.address.encrypt(Deposit(ChunkAddress))
+//!     c. self.address.encrypt(Spend(ChunkAddress))
+//!    And when the register has used 1023 entries:
+//!     d. self.address.encrypt(Extend(RegisterAddress))
+//!     ... which would occupy the last entry, and thus link to a new register.
+//! 3. For a private Wallet:
+//!     a. Store a register with address of self.address.encrypt(self.address).
+//!     ... then follow from b. in 2.
+//! 4. Then, when a wallet is to be loaded from the network:
+//!     a. Get the `PublicAddress` from your secret.
+//!     b. Fetch the register with address of either the plaintext of or the encrypted `PublicAddress`.
+//!     c. Decrypt all entries and apply the ops to your Wallet, to get the current state of it.
+//!     d. If there is another register linked at the end of this one, follow that link and repeat steps b., c. and d.
+//!
+//! We will already now pave for that, by mimicing that flow for the local storage of a Wallet.
+//! First though, a simpler local storage will be used. But after that a local register store can be implemented.
+
+mod error;
+mod local_store;
+mod network_store;
+
+pub use self::{
+    error::{Error, Result},
+    local_store::{LocalDepositWallet, LocalSendWallet},
+    // network_store::NetworkWallet,
```

Comment:
Remove this comment?

Reply:
Yeah, could do. That's the next iteration stuff.